frontend/spec/features/product_prices_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'spec_helper'
+
+describe 'Product with prices in multiple currencies', type: :feature, js: true do
+  context 'with USD, EUR and GBP as currencies' do
+    let!(:product) { create(:product) }
+
+    before do
+      reset_spree_preferences do |config|
+        config.supported_currencies   = 'USD,EUR,GBP'
+        config.allow_currency_change  = true
+        config.show_currency_selector = true
+      end
+      create(:price, variant: product.master, currency: 'EUR', amount: 16.00)
+      create(:price, variant: product.master, currency: 'GBP', amount: 23.00)
+    end
+
+    it 'can switch by currency', :js do
+      visit spree.product_path(product)
+      expect(page).to have_text '$19.99'
+      select 'EUR', from: 'currency'
+      expect(page).to have_text '€16.00'
+      select 'GBP', from: 'currency'
+      expect(page).to have_text '£23.00'
+    end
+
+    context 'and :show_currency_selector is false' do
+      before do
+        reset_spree_preferences do |config|
+          config.supported_currencies   = 'USD,EUR,GBP'
+          config.allow_currency_change  = true
+          config.show_currency_selector = false
+        end
+      end
+
+      it 'will not render the currency selector' do
+        visit spree.product_path(product)
+        expect(current_path).to eq(spree.product_path(product))
```

Comment:
Using the `eq` matcher here produces a race condition that can lead to flaky tests. Prefer Capybara provided matchers when they exist

    expect(page).to have_current_path(spree.product_path(product))